[PyTorch] Use Half, not float16_t, in fp16 gemv fast path signatures

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #139220
* #139208
* #139159
* #139090
* #139084
* #139083
* #139082
* #139081
* #138275
* #138005
* #137918
* #137917
* #137916
* #137915
* #137914
* __->__ #137913
* #137912
* #137911
* #137661

float16_t is ARM-specific. Half is not.

Differential Revision: [D64218427](https://our.internmc.facebook.com/intern/diff/D64218427/)

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

swolchok committed:
[PyTorch] Use Half, not float16_t, in fp16 gemv fast path signatures

float16_t is ARM-specific. Half is not.

Differential Revision: [D64218427](https://our.internmc.facebook.com/intern/diff/D64218427/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Use Half, not float16_t, in fp16 gemv fast path signatures"


float16_t is ARM-specific. Half is not.

Differential Revision: [D64218427](https://our.internmc.facebook.com/intern/diff/D64218427/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Use Half, not float16_t, in fp16 gemv fast path signatures"


float16_t is ARM-specific. Half is not.

Differential Revision: [D64218427](https://our.internmc.facebook.com/intern/diff/D64218427/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

malfet reviewed (commented):
Is this to make it accessible on x86? Otherwise float16_t feels like a reasonable type, isn't it?

swolchok committed:
Update on "[PyTorch] Use Half, not float16_t, in fp16 gemv fast path signatures"


float16_t is ARM-specific. Half is not.

Differential Revision: [D64218427](https://our.internmc.facebook.com/intern/diff/D64218427/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Use Half, not float16_t, in fp16 gemv fast path signatures"


float16_t is ARM-specific. Half is not.

Differential Revision: [D64218427](https://our.internmc.facebook.com/intern/diff/D64218427/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok commented:
> Is this to make it accessible on x86?

yes

swolchok committed:
fix precision on "[PyTorch] Use Half, not float16_t, in fp16 gemv fast path signatures"


float16_t is ARM-specific. Half is not.

Differential Revision: [D64218427](https://our.internmc.facebook.com/intern/diff/D64218427/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Use Half, not float16_t, in fp16 gemv fast path signatures"


float16_t is ARM-specific. Half is not.

Differential Revision: [D64218427](https://our.internmc.facebook.com/intern/diff/D64218427/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Use Half, not float16_t, in fp16 gemv fast path signatures"


float16_t is ARM-specific. Half is not.

Differential Revision: [D64218427](https://our.internmc.facebook.com/intern/diff/D64218427/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Use Half, not float16_t, in fp16 gemv fast path signatures"


float16_t is ARM-specific. Half is not.

Differential Revision: [D64218427](https://our.internmc.facebook.com/intern/diff/D64218427/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Use Half, not float16_t, in fp16 gemv fast path signatures"


float16_t is ARM-specific. Half is not.

Differential Revision: [D64218427](https://our.internmc.facebook.com/intern/diff/D64218427/)

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]
